cachi2/core/package_managers/gomod.py

```diff
@@ -173,9 +173,9 @@ def _resolve_gomod(path: Path, request: Request, git_dir_path=None):
             "GOMODCACHE": "{}/pkg/mod".format(temp_dir),
         }
 
-        if worker_config.cachito_athens_url:
-            athens_url = worker_config.cachito_athens_url
-            env["GOPROXY"] = f"{athens_url}|{athens_url}"
+        if worker_config.cachito_goproxy_url:
+            goproxy_url = worker_config.cachito_goproxy_url
+            env["GOPROXY"] = f"{goproxy_url}|{goproxy_url}"
```

Comment:
This becomes `<proxy>,direct|<proxy>,direct`, which is some mix of fallback and retry behavior (not sure what mix exactly)

https://github.com/golang/go/issues/37367

If we're going to allow the goproxy_url to be more than a url, then it should probably define the entire content of the GOPROXY variable

And since we already have retry logic in place for the whole `go mod download`, it seems fine to keep just `<proxy>,direct`